File: bin/client/dotatribe/resource/script/view/UIHelper.lua
--Name : UIHelper.lua
--Function : 提供Lua页面基类和页面管理
--Author : fangjun

require "resource/script/util/util_function"
require "resource/script/view/UIWindow"

local uiHelper = {}
uiHelper.__index = uiHelper

log("============UIHelper:LuaGetInstanceName():"..luaUIHelper:LuaGetInstanceName())

local uiHelperCXX = tolua.cast(luaUIHelper:LuaGetInstance(), luaUIHelper:LuaGetInstanceName())

-- 获得Helper实例对象
function Call_UIHelper()
	return uiHelper
end

-- 获得根节点Layer
function uiHelper:GetUILayer()
	return uiHelperCXX:LuaGetUILayer()
end

-- 获得c++提供的绘制节点
function uiHelper:GetProvideNode()
	return uiHelperCXX:LuaGetProvideNode()
end

-- 注册需要加载的ll_xxx_layout.txt
function uiHelper:RegisterViewFile(idx, filename, encyfilename)
	return uiHelperCXX:LuaRegisterViewFile(idx, filename, encyfilename)
end

-- autocreate节点
function uiHelper:AutoCreateNodeByTable(key, parentNode)
	return uiHelperCXX:LuaAutoCreateNodeByTable(key, parentNode)
end

-- 强制清除动画缓存
function uiHelper:ForceClearAnimationCache()
	return uiHelperCXX:LuaForceClearAnimationCache()
end

function uiHelper:GetUIWindowData(windowType)
	return UIWindow[windowType]
end

File: src/project/DOTATribe/src/UIHelper.cpp
//Name: UIHelper.h
//Function: 提供Lua使用UISystem中的部分功能接口
//Date: 20140313
//Author: fangjun

#include "../include/UIHelper.h"
#include "../include/UISystem.h"
#include "../include/UIViewDataManger.h"
#include "../include/DOTATribeApp.h"

BaseHelper * luaUIHelper::m_pHelper = NULL;
std::string luaUIHelper::m_sName = "";

luaUIHelper::luaUIHelper()
{
	m_pProvideNode = NULL;
}

luaUIHelper::~luaUIHelper()
{
	m_pProvideNode = NULL;
}

void luaUIHelper::SetProvideNode(CCNode * node)
{
	m_pProvideNode = node;
}

void luaUIHelper::BuildInstance(std::string name, BaseHelper * helper)
{
	m_pHelper = helper;
	m_sName = name;
}

void luaUIHelper::LoadViewFile()
{
	for(unsigned int i = 0; i < m_vRegisterViewFile.size(); ++i)
	{
		UIVIEWDATAMANAGER->registerFile(m_vRegisterViewFile[i].idx, m_vRegisterViewFile[i].filename, m_vRegisterViewFile[i].encryfilename);
	}
}

BaseHelper * luaUIHelper::LuaGetInstance()
{
	return m_pHelper;
}

std::string luaUIHelper::LuaGetInstanceName()
{
	return m_sName;
}

CCLayer * luaUIHelper::LuaGetUILayer()
{
	return UISYSTEM->GetUILayer();
}

CCNode * luaUIHelper::LuaGetProvideNode()
{
	CCNode * temp = m_pProvideNode;
	m_pProvideNode = NULL;

	return temp;
}

void luaUIHelper::LuaAutoCreateNodeByTable(int key, CCNode* parentNode)
{
	return UIVIEWDATAMANAGER->AutoCreateNodeByTable(key, parentNode);
}

void luaUIHelper::LuaRegisterViewFile(int idx, std::string filename, std::string Encyfilename)
{
	UIHelper_ViewFile a(idx, filename, Encyfilename);
	m_vRegisterViewFile.push_back(a);
}

void luaUIHelper::LuaForceClearAnimationCache()
{
	DOTATribeApp::get_instance2()->ForceClearAnimationCache();
}